code/CronTaskStatus.php

```diff
@@ -32,23 +42,60 @@ public static function get_status($class) {
 	 *
 	 * @param string $class Name of class which implements CronTask
 	 * @param bool $wasRun Flag indicating that the task was run this request
+	 * @param string $status Flag indicating new status of the task
 	 * @return CronTaskStatus Status data object
 	 */
-	public static function update_status($class, $wasRun) {
+	public static function update_status($class, $wasRun, $status = null) {
 		// Get existing object
 		$object = static::get()
 			->filter('TaskClass', $class)
 			->first();
 		// Create new object if not found
 		if(!$object) {
-			$object = static::create();
-			$object->TaskClass = $class;
+			$object = $object = static::register_task($class);
 		}
 		// Update fields
 		$now = SS_Datetime::now()->getValue();
 		if($wasRun) $object->LastRun = $now;
+		if($status) $object->Status = $status;
 		$object->LastChecked = $now;
 		$object->write();
 		return $object;
 	}
+
+	/**
+	 * Register new task
+	 *
+	 * By default all new tasks are in status 'On' and
+	 * will use default schedule string
+	 *
+	 * @param string $class
+	 * @return static
+	 */
+	public static function register_task($class) {
+		$object = static::create();
+		$inst = singleton($class);
```

Comment:
Do we need to instantiate a singleton for the purpose of getting the schedule? can't we just use `$object`?